### PR TITLE
Implement pagination using auto_paginate

### LINKED ISSUE #9: Handle incomplete results from GitHub API
Currently, incomplete results (i.e. queries that return > 100 repositories) will just generate a warning.  The remaining repositories beyond 100 will be ignored.

### PR DESCRIPTION
**Pull request by duncan-bayne**
It turns out there's no point in manually paginating; the Oktokit docs recommend the use of `auto_paginate` instead.

This PR changes to using auto_paginate, and introduces an integration test with HTTP record and replay courtesy VCR.  In addition to the automated integration test, I've manually verified this against a test account with 200 EOL Ruby repositories in it.

I'm not sure where the upper bound lies for auto-pagination :)

It also:

- Adds some development dependencies to the Gemspec.
- Bumps Octokit to 4.22.
- Adds style checks with standard to the default Rake task.

Closes #9.

### CODE REVIEW

File: end_of_life.gemspec [hunk 1]
```diff
@@ -32,12 +32,18 @@ Gem::Specification.new do |spec|
...
+  spec.add_dependency "octokit", "~> 4.22"
+  spec.add_dependency "pastel", "~> 0.8.0"
   spec.add_dependency "tty-spinner", "~> 0.9.0"
   spec.add_dependency "tty-table", "~> 0.12.0"
 
+  spec.add_development_dependency "pry", "~> 0.14"
+  spec.add_development_dependency "rspec", "~> 3.10"
+  spec.add_development_dependency "rspec-mocks", "~> 3.10"
+  spec.add_development_dependency "vcr", "~> 6.0.0"
+  spec.add_development_dependency "webmock", "~> 3.13"
```
Comment (MatheusRich) [inline end_of_life.gemspec:45]: Could you move these to `Gemfile`?

File: end_of_life.gemspec [hunk 2]
```diff
@@ -32,12 +32,18 @@ Gem::Specification.new do |spec|
...
 
-  spec.add_dependency "octokit", "~> 4.0"
-  spec.add_dependency "pastel", "~> 0.8.0"
   spec.add_dependency "dry-monads", "~> 1.3"
+  spec.add_dependency "octokit", "~> 4.22"
+  spec.add_dependency "pastel", "~> 0.8.0"
   spec.add_dependency "tty-spinner", "~> 0.9.0"
   spec.add_dependency "tty-table", "~> 0.12.0"
 
+  spec.add_development_dependency "pry", "~> 0.14"
```
Comment (MatheusRich) [inline end_of_life.gemspec:41]: I've been using `binding.irb` for a while, and it served me well. Can we remove pry for now?

File: spec/end_of_life/repository_integration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  describe "#fetch" do
+    context "given an account with 200 end of life repositories" do
+      subject(:repositories) do
+        VCR.use_cassette("many_repositories") do
+          EndOfLife::Repository.fetch(language: "ruby", user: nil, organizations: nil, repository: nil)
+        end
+      end
+
+      it "returns Success with the collection of repositories" do
+        expect(repositories.value_or(nil).count).to eq(200)
```
Comment (MatheusRich) [inline spec/end_of_life/repository_integration_spec.rb:16]: If you're confident that the response is a Success, you can use `value!` to extract it out of the monad

```suggestion
        expect(repositories.value!.count).to eq(200)
```

File: spec/end_of_life/repository_spec.rb [hunk 4]
```diff
@@ -108,6 +107,41 @@
       end
     end
 
+    describe "#fetch" do
+      let(:client) { instance_double(Octokit::Client, :user => user) }
```
Comment (MatheusRich) [inline spec/end_of_life/repository_spec.rb:111]: Do you mind updating the tests to not use `let`s and possibly `before`s? I'm [not a big fan](https://thoughtbot.com/blog/lets-not) of them 😅

File: end_of_life.gemspec (hunk 1 above)
Comment (duncan-bayne) [reply end_of_life.gemspec]: Done :)

File: end_of_life.gemspec (hunk 2 above)
Comment (duncan-bayne) [reply end_of_life.gemspec]: Done :)

File: spec/end_of_life/repository_integration_spec.rb (hunk 3 above)
Comment (duncan-bayne) [reply spec/end_of_life/repository_integration_spec.rb]: Done :)

File: spec/end_of_life/repository_spec.rb [hunk 5]
```diff
@@ -108,6 +107,41 @@
...
+          allow(client).to receive(:search_repositories).and_return(response)
+        end
+
+        it "calls the GitHub API once" do
+          repositories
+          expect(client).to have_received(:search_repositories).once
+        end
+
+        it "returns Success with the collection of repositories" do
+          expect(repositories.value_or(nil).count).to eq(1)
```
[the remark this answers is not in the document]
Comment (duncan-bayne) [reply spec/end_of_life/repository_spec.rb]: Done :)

File: spec/end_of_life/repository_spec.rb (hunk 4 above)
Comment (duncan-bayne) [reply spec/end_of_life/repository_spec.rb]: Rewritten, & broken out into two `it`s.

File: spec/end_of_life/repository_integration_spec.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require "climate_control"
+require "ostruct"
+
+RSpec.describe EndOfLife::Repository do
+  describe "#fetch" do
+    it "fetches all 200 repositories from an account despite exceeding page size" do
+      repositories = VCR.use_cassette("many_repositories") do
```
Comment (MatheusRich) [inline spec/end_of_life/repository_integration_spec.rb:9]: Looks like this is not using the VCR cassette. Make sure to commit the cassettes if you want CI to use them.

Be sure that the cassettes don't leak your GITHUB_TOKEN. If they do, we can just rely on mocks, instead.

Comment (duncan-bayne) [reply spec/end_of_life/repository_integration_spec.rb:9]: Ah!  Got it.  Amazing what a few hours of good sleep can do ...

It _is_ using the cassette, but the check for the presence of GITHUB_TOKEN happens way before the cassette is read.  The fix is to set the environment variable in the spec.

Thanks for the warning re. leaking real tokens.  I've taken care of this with the following configuration:

`config.filter_sensitive_data("REDACTED") { ENV["GITHUB_TOKEN"] }`

**Comment by MatheusRich** [comment]
Awesome! Thanks for the contribution, @duncan-bayne! 🎉

**Pull request merged**